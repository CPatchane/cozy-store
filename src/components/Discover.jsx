import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
