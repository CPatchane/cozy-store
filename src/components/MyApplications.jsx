import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
