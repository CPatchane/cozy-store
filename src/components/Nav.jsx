import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
