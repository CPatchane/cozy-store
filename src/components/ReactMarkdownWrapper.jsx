import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
