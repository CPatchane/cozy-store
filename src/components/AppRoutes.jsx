import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
