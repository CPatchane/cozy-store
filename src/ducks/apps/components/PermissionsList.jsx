/* global cozy */

import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import ReactMarkdownWrapper from '../../components/ReactMarkdownWrapper'

export const PermissionsList = ({ t, permissions, appName }) => {
  const permissionsArray = Object.values(permissions).map(p => {
    p.typeDescription = t(`doctypes.${p.type}`).replace(/^doctypes\./, '')
    return p
  })
  return permissionsArray.length ? (
    <div>
      <ReactMarkdownWrapper
        source={t('app_modal.install.permissions.description', {
          cozyName: cozy.client._url.replace(/^\/\//, ''),
          appName
        })}
      />
      <ul className='sto-perm-list'>
        {permissionsArray.map(permission => (
          <li key={permission.type} className={permission.type}>
            <ReactMarkdownWrapper
              source={`__${permission.typeDescription}__`}
            />
          </li>
        ))}
      </ul>
    </div>
  ) : (
    <ReactMarkdownWrapper
      source={t('app_modal.install.permissions.nothing', {
        cozyName: cozy.client._url.replace(/^\/\//, ''),
        appName
      })}
    />
  )
}

export default translate()(PermissionsList)
