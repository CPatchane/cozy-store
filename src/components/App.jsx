
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
