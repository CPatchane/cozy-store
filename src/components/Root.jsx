import React from 'react'
import { Provider } from 'react-redux'
import { I18n } from 'cozy-ui/react/I18n'
import { HashRouter } from 'react-router-dom'

import App from './App'

const Root = ({ context, lang, store }) => {
  return <I18n lang={lang} dictRequire={(lang) => require(`../locales/${lang}`)}>
    <Provider store={store}>
      <HashRouter>
        <App />
      </HashRouter>
    </Provider>
  </I18n>
}

export default Root
