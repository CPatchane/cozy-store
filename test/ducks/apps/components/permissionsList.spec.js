'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../jestLib/I18n'
import { PermissionsList } from 'ducks/apps/components/PermissionsList'

import mockAppVersion from '../_mockPhotosRegistryVersion'
import mockKonnectorVersion from '../_mockPKonnectorTrinlaneRegistryVersion'

Enzyme.configure({ adapter: new Adapter() })

describe('PermissionsList component', () => {
  beforeAll(() => {
    // define global mock url
    global.cozy = {
      client: {
        _url: '//cozytest.mock.cc'
      }
    }
  })

  it('should be rendered correctly with permissions', () => {
    const component = shallow(
      <PermissionsList t={tMock} app={mockAppVersion.manifest} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with konnector permissions', () => {
    const component = shallow(
      <PermissionsList t={tMock} app={mockKonnectorVersion.manifest} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly without permissions', () => {
    const component = shallow(
      <PermissionsList t={tMock} app={{ name: 'Mock' }} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
