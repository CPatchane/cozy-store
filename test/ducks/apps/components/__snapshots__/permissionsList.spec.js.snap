// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApplications component should be rendered correctly with konnector permissions 1`] = `
<div>
  <ReactMarkdownWrapper
    source="In order to improve the value of your Cozy (cozytest.mock.cc), the application Mock requires access to the following data:"
  />
  <ul
    className="sto-perm-list"
  >
    <li
      className="io.cozy.bank.operations"
    >
      <ReactMarkdownWrapper
        source="__io.cozy.bank.operations__"
      />
    </li>
    <li
      className="io.cozy.bills"
    >
      <ReactMarkdownWrapper
        source="__Bills__"
      />
    </li>
    <li
      className="io.cozy.files"
    >
      <ReactMarkdownWrapper
        source="__Files__"
      />
    </li>
    <li
      className="io.cozy.accounts"
    >
      <ReactMarkdownWrapper
        source="__Accounts__"
      />
    </li>
  </ul>
</div>
`;

exports[`MyApplications component should be rendered correctly with permissions 1`] = `
<div>
  <ReactMarkdownWrapper
    source="In order to improve the value of your Cozy (cozytest.mock.cc), the application Mock requires access to the following data:"
  />
  <ul
    className="sto-perm-list"
  >
    <li
      className="io.cozy.files"
    >
      <ReactMarkdownWrapper
        source="__Files__"
      />
    </li>
    <li
      className="io.cozy.apps"
    >
      <ReactMarkdownWrapper
        source="__List of installed applications__"
      />
    </li>
    <li
      className="io.cozy.photos.albums"
    >
      <ReactMarkdownWrapper
        source="__Photo albums__"
      />
    </li>
    <li
      className="io.cozy.contacts"
    >
      <ReactMarkdownWrapper
        source="__Contacts__"
      />
    </li>
    <li
      className="io.cozy.settings"
    >
      <ReactMarkdownWrapper
        source="__Settings__"
      />
    </li>
  </ul>
</div>
`;

exports[`MyApplications component should be rendered correctly without permissions 1`] = `
<ReactMarkdownWrapper
  source="The application Mock doesn't require any data access to work correctly on your Cozy (cozytest.mock.cc)."
/>
`;
